### DISCUSSION TITLE
Integrate FluxCD monitoring with Dynatrace

### DISCUSSION THREAD

**Discussion by TOHUHM** [fluxcd/flux2#3792] [Q&A]
Hello 

Flux monitoring with Prometheus is perfect, we can get the metrics of FluxCD controller easily, but now we moved to Dynatrace due to some internal policy, I searched in internet, there are almost no information about how to monitor FluxCD with Dynatrace, I want to know is there any best practice to monitor Flux with Dynatrace?  Thanks a lot！

**Comment by grabnerandi** [comment]
Hi. There are several ways on how you can ingest your data into Dynatrace. For K8s the best is to simply install the Dynatrace Operator. If you follow the instructions your K8s cluster will automatically be monitored by Dynatrace giving you also the option to enable automated Prometheus scraping - so - all your prometheus metrics are also automatically in Dynatrace.
If you cant install the Dynatrace Operator you can also use the Dynatrace Prometheus extension or use e.g: OpenTelemetry collector to send the metrics to Dynatrace.

I also have the following YouTube Video that shows how to monitor K8s with Dynatrace at scale: https://www.youtube.com/watch?v=j-Zc_ck-Orc

**Reply by TOHUHM** [reply]
Hello Grabnerandi
Thanks for your help, It's really helpful, we have got the gotk metric, by the way,  I want to make the same dashboard as Grafana which provided here: https://fluxcd.io/flux/guides/monitoring/
is there any easy solution to transfer granfa dashboard to dynatrace dashboard, as we already there are some function like "rate", which is not available in dynatrace.
For OpenTelemetry, I also want to know how it integrate with Flux Controller.

Thanks a lot!

**Reply by stefanprodan** [reply]
The Flux controllers emit Kubernetes events, I don't know if Dynatrace can capture those but other solutions like Grafana Loki do. It is recommend to capture both metrics and events to properly monitor Flux.

**Reply by TOHUHM** [reply]
Hello Stefanproda

Yes,  dynatrace can capture the Flux event from kubernetes events, I can see the events in dynatrace UI already, It's another dimension I'm working on.
Here I am looking for a solution to transfer the "expression" in "https://github.com/fluxcd/flux2/blob/main/manifests/monitoring/monitoring-config/dashboards/cluster.json"  to dynatrace.
of course we can study the expression one by one to integrate it with dynatrace dashboard, but it's painful when we met some prometheus calculate method like "rate"..  
Also It would be great if Flux officially provide "monitoring with dynatrace" wiki.

**Reply by stefanprodan** [reply]
> Also It would be great if Flux officially provide "monitoring with dynatrace" wiki.

This is not a Flux OSS concern. If Dynatrace whats to be appealing to Flux users, they should do what Datadog did here https://github.com/fluxcd/flux2/discussions/3157

**Reply by grabnerandi** [reply]
Hi. Thanks for the discussion here. As mentioned - Dynatrace already by default can scrape metrics, events and also auto-ingest logs. TJhats why I dont think we need to build an additional integration. What would be good to know is which Flux metrics are important and based on that we can easily create a Dynatrace dashboard. What doesnt exist is a Grafana dashboard converter. But - as Dynatrace has an easy to use dashboarding UI you should be able to quickly put this together. @TOHUHM - I suggest you bring this discussion to the dynatrace community as I am sure there are others that are trying to do the same thing -> https://community.dynatrace.com

**Reply by Thilaknath** [reply]
@TOHUHM Were you able to find a solution to your problem. We are on a similar situation trying to build dynatrace dashboard for flux metrics. Any guidance here would be appreciated.
